static/js/containers/PostPage.js

```diff
@@ -83,13 +84,23 @@ class PostPage extends React.Component {
   }
 
   render() {
-    const { posts, channels, comments, forms } = this.props
+    const { posts, channels, comments, forms, dispatch } = this.props
     const [postId, channelName] = this.getMatchParams()
+    const dispatchableToggleUpvote = toggleUpvote(dispatch)
 
     return (
       <Loading
         restStates={[posts, channels, comments]}
-        renderContents={() => this.renderContents(postId, posts.data, channelName, channels.data, comments.data, forms)}
+        renderContents={() =>
+          this.renderContents(
+            postId,
+            posts.data,
+            channelName,
+            channels.data,
+            comments.data,
+            forms,
+            dispatchableToggleUpvote
+          )}
```

Comment:
The parameter passing here is really bulky. Let's just bind `this` to the handler function. Rewritten, this would like like: 

```javascript
renderContents() {
    const { posts, channels, comments, forms, dispatch } = this.props
    const [postId, channelName] = this.getMatchParams()
    const dispatchableToggleUpvote = toggleUpvote(dispatch)

    const post = posts.data.get(postId)
    const channel = channels.data.get(channelName)
    const commentTreeData = comments.data.get(postId)

    if (R.isNil(channel) || R.isNil(post) || R.isNil(commentTreeData)) {
      return null
    }
    return (
      <div className="double-column">
        <ChannelBreadcrumbs channel={channel} />
        <div className="first-column">
          <Card>
            <PostDisplay post={post} toggleUpvote={dispatchableToggleUpvote} expanded />
            <ReplyToPostForm forms={forms} post={post} />
          </Card>
          <CommentTree comments={commentTreeData} forms={forms} />
        </div>
      </div>
    )
  }

  render() {
    const { posts, channels, comments } = this.props
    return (
      <Loading
        restStates={[posts, channels, comments]}
        renderContents={this.renderContents.bind(this)}
      />
    )
  }
```

The same idea can also be applied to `ChannelPage.js`.